src/_vendor/zod-to-json-schema/parsers/optional.ts

```diff
@@ -3,6 +3,15 @@ import { JsonSchema7Type, parseDef } from '../parseDef';
 import { Refs } from '../Refs';
 
 export const parseOptionalDef = (def: ZodOptionalDef, refs: Refs): JsonSchema7Type | undefined => {
+  if (refs.openaiStrictMode) {
+    const fieldName = refs.propertyPath?.slice(-1)[0] || 'unknown';
+    console.warn(
+      `Warning: Field "${fieldName}" uses .optional() which is not supported by OpenAI API Structured Outputs. ` +
+        `Please use .nullable() instead. ` +
+        `See: https://platform.openai.com/docs/guides/structured-outputs#all-fields-must-be-required`,
+    );
+  }
+
```

Comment:
Unnecessary use of string concatenation (+) since we are already using template literals.